### ISSUE TITLE
[doc,site] Add succinct 'human-name' field to all .hjson

### ISSUE THREAD

**Issue by hcallahan-lowrisc** [lowRISC/opentitan#17598]
### Description

In the initial [earlgrey-diagram](https://opentitan.org/documentation/index.html) PR (https://github.com/lowRISC/opentitan/pull/17405#discussion_r1132433916), it was noted that the diagram tooltips use custom names for each block, and it would be nice to pull this data from a single source of truth. The block level .hjson field 'human-name' would be ideal, but some blocks are currently missing this label, and some have labels which are too long to easily integrate into a tooltip.

**Comment by johngt** [comment]
@hcallahan-lowrisc - based on the comments thread this now looks like it has been merged in and completed?
There was also a short name set captured in https://github.com/lowRISC/opentitan/pull/17286 - believe this is referenced implicitly @andreaskurth ?

**Comment by andreaskurth** [comment]
I think the following tasks are open here:
- Add `human_name` to the remaining HW blocks. Tracked in #17307.
- Add an abbreviation to each HW block. Not tracked in a separate issue yet, could be part of this issue since we currently need it only here.
- Change the diagram generator to use the human-readable name or the abbreviation for each HW block (in the label and in the tooltips; select between name or abbreviation based on some length limitation). Tracked by this issue.

**Comment by hcallahan-lowrisc** [comment]
I think we need to come up with a plan of which .hjson field is used for tooltips. This would probably be well-constrained by imposing a hard character limit. My first thought would be 30 chars, but open for suggestions here.

I see two options for how to do this. 
1. Modify the 'human-name' fields introduced in https://github.com/lowRISC/opentitan/pull/17286 to be strictly less than the character limit.
2. Add a new field (e.g. 'succint-name', 'tooltip-name' or 'abbreviated-name') which meets this character constraint.

After a decision on this, we could then create a task to update the diagram build to pull data from the relevant field.

**Comment by hcallahan-lowrisc** [comment]
I agree that a new 'abbreviation' field makes the most sense. Let's run with that.

Maybe we should consider updating there schema/comments for each fields, as this would bring us to 4 different name or description fields! It would be nice to avoid duplication and be consistent where possible.
https://github.com/lowRISC/opentitan/blob/e2dc6dd5630c0ad189ff5b469272a2a86b088c26/util/reggen/ip_block.py#L58-L61

**Comment by andreaskurth** [comment]
We would just add a new optional field `'abbreviation': ['s', "abbreviated name of the component"]`, no? Where do you see duplication or inconsistencies?

**Comment by andreaskurth** [comment]
> It's the duplication of the content which I think might be a bit messy, for instance...
> 
> ```
> + 'abbreviation' : 'SPI_Device'
> ```

That's why I'd make the `abbreviation` field optional. Some blocks simply cannot be meaningfully abbreviated. Though "SPI Dev" may be a candidate for this example.

In general, I'd avoid using underscores and the like in acronyms. The technical name is `spi_device`, and I think we should use precisely that when technical accuracy is required

**Comment by hcallahan-lowrisc** [comment]
> In general, I'd avoid using underscores and the like in acronyms and other designators meant for humans. The technical name is `spi_device`, and I think we should use precisely that when technical accuracy is required

Yeah so in terms of a schema I think we need to capture some more information like this. I think it would be nice to have more comments/requirements in the 'reggen/ip_block.py' definition file, to encourage the more-constrained use of each field. Comments such as 'only use the `abbreviation` field if the `human-name` cannot be kept below 30 chars', 'avoid using underscores/punctuation in this field, use full prose' etc. Too many fields without adequate constraints will make it easy to duplicate information. 
I'll try and put together a PR of what this might look like.

**Comment by andreaskurth** [comment]
> > In general, I'd avoid using underscores and the like in acronyms and other designators meant for humans. The technical name is `spi_device`, and I think we should use precisely that when technical accuracy is required
> 
> Yeah so in terms of a schema I think we need to capture some more information like this. I think it would be nice to have more comments/requirements in the 'reggen/ip_block.py' definition file, to encourage the more-constrained use of each field. Comments such as 'only use the `abbreviation` field if the `human-name` cannot be kept below 30 chars', 'avoid using underscores/punctuation in this field, use full prose' etc. Too many fields without adequate constraints will make it easy to duplicate information. I'll try and put together a PR of what this might look like.

I agree about the requirements/constraints. Please see https://github.com/lowRISC/opentitan/pull/17286 for a list of requirements / constraints / design considerations that went into the existing `human_name` values
